src/components/Filters/Filters.tsx

```diff
@@ -60,7 +60,7 @@ function Filters<F, G extends Value = string>(props: FilterProps<F, G>) {
         getOptionValue={(o) => o.id}
         getOptionLabel={(o) => o.name}
         value={groupBy.value}
-        onSelect={(g) => groupBy.setValue(g)}
+        onSelect={(g) => g && groupBy.setValue(g)}
```

Comment:
Cool, fwiw this is the sort of usage that I was worried/anticipating would be tripped by "now the `value` might be `undefined`".

Granted, we are not setting `unsetLabel` here.

The really fancy version of this PR would try to take these lines:

```
export function BoundSelectField<T, V extends Value>(props: BoundSelectFieldProps<T, V>): JSX.Element;
export function BoundSelectField<T extends HasIdAndName<V>, V extends Value>(
  props: Optional<BoundSelectFieldProps<T, V>, "getOptionLabel" | "getOptionValue">,
): JSX.Element;
```

Which are providing overloads of `SelectField` that saying "oh, your options already have id and name? np I'll implement getOptionLabel/Value for you" and do something similar like: "oh, you _don't_ set `unsetLabel`?, then you can do just `onSelect(value: V)`".

So then callers like this one, who aren't opting in to the `unsetLabel` "extra option", don't need to know about the "extra option" being passed to their `onSelect`. (This imo is part of why the current approach had been working out, in that sure the caller had to know "inject an extra option", but it kept a strict mapping of "your onSelect(V) and options/values always line up 1-1").

So the fancier typing is...probably...doable, but more complex that I really want to chase down at this point, so just mentioning, but this is fine.

Reply:
I see what you mean. That would be nice to infer the user didn't include the `unsetLabel` and change the typing of `onSelect`. Again, I appreciate the thorough and thoughtful feedback.